### PR TITLE
Ansible: SQL Users

### PR DESCRIPTION
**Pull request by rambleraptor**
User support for SQL on Ansible

- This also has some cleanup + refactors. Mostly in relation to fetch_wrapped_resource, which is a SQL user thing.

-----------------------------------------------------------------
# [all]
Ansible: SQL Users + fetch_wrapped_resource support
## [ansible]
Ansible: SQL Users

### CODE REVIEW

File: products/sql/ansible.yaml [hunk 1]
```diff
@@ -76,15 +76,39 @@ overrides: !ruby/object:Provider::ResourceOverrides
           module.fail_json(msg="Incorrect result: {kind}".format(**result))
 
       return result
+  User: !ruby/object:Provider::Ansible::ResourceOverride
+    unwrap_resource: true
+    return_if_object: |
```
Comment (danawillow) [inline products/sql/ansible.yaml]: I don't love that every object in SQL needs the same code copy/pasted here. Is there a way to factor it out somehow?

File: products/sql/examples/ansible/user.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+--- !ruby/object:Provider::Ansible::Example
+dependencies:
+  - !ruby/object:Provider::Ansible::Task
+    name: gcp_sql_instance
+    code: |
+      name: <%= dependency_name('instance', 'user') %>
```
Comment (danawillow) [inline products/sql/examples/ansible/user.yaml]: nit: We're supposed to use `-%>` everywhere, right?

File: templates/ansible/resource.erb [hunk 3]
```diff
@@ -274,15 +274,39 @@ def resource_to_request(module):
             return_vals[k] = v
 
     return return_vals
+<% unless false?(object.unwrap_resource) -%>
```
Comment (danawillow) [inline templates/ansible/resource.erb]: I'm going to go ahead and trust your judgment here but to my still-new-at-ruby-eye, this is less readable than `if object.unwrap_resource`

File: templates/ansible/resource.erb [hunk 4]
```diff
@@ -274,15 +274,39 @@ def resource_to_request(module):
...
+    'return {',
+    indent_list(
+      Hash[object.identity.map { |i| [i, "module.params[#{quote_string(i.out_name)}]"] }]
+        .map { |k, v| "#{quote_string(k.out_name)}: #{v}" }, 4
+    ),
+    '}'
+  ]
+-%>
+<%= lines_before(lines(emit_method('unwrap_resource_filter', %w[module],
+                             urf_code, file_relative), 1), 1) -%>
```
Comment (danawillow) [inline templates/ansible/resource.erb]: nit: is this the correct indentation?

File: templates/ansible/resource.erb (hunk 3 above)
Comment (rambleraptor) [reply templates/ansible/resource.erb]: This is something we should bring up for a potential bug fix-it.

In Ruby, we have true, false, and nil.

nil == false, but false != nil.

In this case, we want true false, but we don't want to have nils confused in it as well. That's what we made the false? function for.

I'm not a huge fan of this. I know the Ruby community must have a better way to fix this.

File: products/sql/examples/ansible/user.yaml (hunk 2 above)
Comment (rambleraptor) [reply products/sql/examples/ansible/user.yaml]: Yeah. This is a bug that I want to address as part of a bigger rewrite of examples.

File: templates/ansible/resource.erb (hunk 4 above)
Comment (rambleraptor) [reply templates/ansible/resource.erb]: Good find! Addressed.

**Pull request merged**